src/json.cc

```diff
@@ -224,7 +226,35 @@ Obj JSON_ESCAPE_STRING(Obj self, Obj param)
     return copy;
 }
 
+Obj GAP_PLIST_TO_JSON_STRING(Obj self, Obj string, Obj stream, Obj list) {
+    Int len = LEN_PLIST(list);
+    char buf[50] = {};
+    Int first = 1;
+
+    // Call this at the start
+    ConvString(string);
+    AppendCStr(string, "[", 1);
```

Comment:
`AppendCStr` is only available in GAP >= 4.12. If you'd like to stay compatible with older versions, I would suggest to simply copy the code from the GAP kernel and then slightly rename it (`JSON_AppendCStr`?).